### PR TITLE
User moods

### PR DESCRIPTION
**Pull request by IgorrMagalhaess**
## Type of change
- [x] New feature
- [ ] Bug Fix
- [x] Test
- [ ] Refactor/Style

## Description
Relevant User Story(s):
*Brief summary of the change(s) included in this pull request*
 - User Moods microservice with endpoints to GET all the moods for a user and POST to create mood for a user.
 - Controller, Model, Poros and request fully tested, 100% coverage, all tests passing.
 - Tests for bad requests and edge cases
 - PR template
 - README with endpoints and how to use the app

This description closes #

## Check the correct boxes
- [x] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

## Testing Changes
- [ ] No Tests have been changed
- [ ] Some Tests have been changed
- [x] All of the Tests have been changed

*Please describe what in the world happened if tests are broken*

## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

## Fun fact or favorite .gif:
The term "bug" to describe a glitch in a software program originated in 1947 when Grace Hopper, a computer scientist, found an actual moth causing issues in the Harvard Mark II computer. She famously removed the moth and taped it into the logbook with the note: "First actual case of bug being found." This incident led to the term "debugging" to describe the process of fixing issues in software.

### CODE REVIEW

File: app/controllers/api/v1/moods_controller.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      avg_mood: average_mood
+    }
+    render json: MoodSerializer.new(mood_summary).serialize
+  end
+
+  def create
+    begin
+      mood = Mood.create!(mood_params)
+      head :created
+    rescue => e
```
Comment (JackCSweeney) [inline app/controllers/api/v1/moods_controller.rb]: Should we look into abstracting this to the application controller for the error handling at a later, refactor, point? Def not necessary, but might be nice to slim down the controller a little bit!

File: app/models/mood.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  def self.user_moods(user_id)
+    where(user_id: user_id)
+  end
+
+  def self.average_mood(user_id)
+    avg = where(user_id: user_id).average(:mood).to_f.round(1)
+  end
+
+  def set_date
+    self.date ||= Time.zone.now.strftime("%Y-%m-%d")
```
Comment (JackCSweeney) [inline app/models/mood.rb:17]: Interesting, I was figuring we could just use the 'created at' column that gets added to each record on creation as our way of keeping track of the dates that a user input their mood, but this works too and makes the info a little more human readable without having to convert/change anything later!

File: app/serializers/mood_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  def serialize
+    {
+      data: {
+        id: @mood_summary[:user_id],
+        type: 'user_mood',
+        attributes: {
+          avg_mood: @mood_summary[:avg_mood],
+          user_moods: @mood_summary[:user_moods].map { |mood| { date: mood.date, mood: mood.mood } }
```
Comment (JackCSweeney) [inline app/serializers/mood_serializer.rb:15]: This looks great! And I can see why using that model method I commented on for the date makes more sense here!

File: spec/requests/api/v1/user_moods_spec.rb [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+      expect(response.status).to eq 400
+
+      mood_response = JSON.parse(response.body, symbolize_names: true)
+
+      expect(mood_response).to be_a(Hash)
+      expect(mood_response[:errors]).to be_a(Array)
+      expect(mood_response[:errors].first[:detail]).to eq("Validation failed: Mood is not a number")
+    end
+  end
+end
```
Comment (JackCSweeney) [inline spec/requests/api/v1/user_moods_spec.rb:174]: Great testing here! Love the thoroughness!

**Review by JackCSweeney — APPROVED** [review]
Looks fantastic to me Igor!

File: app/controllers/api/v1/moods_controller.rb (hunk 1 above)
Comment (IgorrMagalhaess) [reply app/controllers/api/v1/moods_controller.rb]: Oh this is easy I can do this before merging
